build/akri-containers.mk

```diff
@@ -58,339 +46,17 @@ ifeq (1, ${BUILD_ARM64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/onvif-video-broker:$(LABEL_PREFIX) $(PREFIX)/onvif-video-broker:$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
-endif
-
-opcua-monitoring-docker-multi-arch-create:
-ifeq (1, ${BUILD_AMD64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/opcua-monitoring-broker:$(LABEL_PREFIX) $(PREFIX)/opcua-monitoring-broker:$(LABEL_PREFIX)-$(AMD64_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM32})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/opcua-monitoring-broker:$(LABEL_PREFIX) $(PREFIX)/opcua-monitoring-broker:$(LABEL_PREFIX)-$(ARM32V7_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/opcua-monitoring-broker:$(LABEL_PREFIX) $(PREFIX)/opcua-monitoring-broker:$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
-endif
-
-anomaly-detection-docker-multi-arch-create:
-ifeq (1, ${BUILD_AMD64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/anomaly-detection-app:$(LABEL_PREFIX) $(PREFIX)/anomaly-detection-app:$(LABEL_PREFIX)-$(AMD64_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM32})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/anomaly-detection-app:$(LABEL_PREFIX) $(PREFIX)/anomaly-detection-app:$(LABEL_PREFIX)-$(ARM32V7_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/anomaly-detection-app:$(LABEL_PREFIX) $(PREFIX)/anomaly-detection-app:$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
-endif
-
-webhook-configuration-docker-multi-arch-create:
-ifeq (1, ${BUILD_AMD64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/webhook-configuration:$(LABEL_PREFIX) $(PREFIX)/webhook-configuration:$(LABEL_PREFIX)-$(AMD64_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM32})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/webhook-configuration:$(LABEL_PREFIX) $(PREFIX)/webhook-configuration:$(LABEL_PREFIX)-$(ARM32V7_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/webhook-configuration:$(LABEL_PREFIX) $(PREFIX)/webhook-configuration:$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
-endif
-
-streaming-docker-multi-arch-create:
-ifeq (1, ${BUILD_AMD64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/video-streaming-app:$(LABEL_PREFIX) $(PREFIX)/video-streaming-app:$(LABEL_PREFIX)-$(AMD64_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM32})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/video-streaming-app:$(LABEL_PREFIX) $(PREFIX)/video-streaming-app:$(LABEL_PREFIX)-$(ARM32V7_SUFFIX)
-endif
-ifeq (1, ${BUILD_ARM64})
-	$(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/video-streaming-app:$(LABEL_PREFIX) $(PREFIX)/video-streaming-app:$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
-endif
+# Rust targets
+$(eval $(call add_rust_targets,controller,controller))
+$(eval $(call add_rust_targets,agent,agent))
+$(eval $(call add_rust_targets,udev,udev-video-broker))
```

Comment:
To simplify, should we make the convention of having `add_rust_targets` only have one parameter, being the final container tag, and change the `udev` build name to `udev-video-broker`?

Reply:
we could do that.  might require changes to the udev workflow too.  i will check.

Reply:
it looks like there are 4 things with differing makefile target prefixes and dockerfile/container names:

* udev
* onvif
* anomaly-detection
* streaming

Reply:
i think i'll leave that simplification for the next round